src/views/notifications/atlassianNotificationNotifier.ts

```diff
@@ -0,0 +1,243 @@
+                    Logger.warn('notificationFeed is undefined in the response');
+                    return;
+                }
+                response.notifications.notificationFeed.nodes
+                    .filter((node: any) => this.filter(node))
+                    .map((node: any) => {
+                        const notification = this.mapper(authInfo, node);
+                        if (notification) {
+                            if (FeatureFlagClient.checkExperimentValue(Experiments.AtlassianNotifications)) {
+                                NotificationManagerImpl.getInstance().addNotification(notification);
+                            }
+                        }
+                    });
+            })
+            .catch((error) => {
+                Logger.error(error, 'Error fetching notifications');
+            });
+    }
+
+    private mapper(authInfo: AuthInfo, node: any): AtlasCodeNotification | undefined {
+        const product = this.isJiraNotification(node)
+            ? ProductJira
+            : this.isBitbucketNotification(node)
+              ? ProductBitbucket
+              : undefined;
+        if (!product) {
+            Logger.warn(`Unsupported notification type for URL: ${node.headNotification.content.url}`);
+            return undefined;
+        }
+        const notificationType = product === ProductJira ? NotificationType.JiraComment : NotificationType.PRComment;
+
+        // Strip query parameters from the URL before creating the Uri
+        const url = node.headNotification.content.url.split('?')[0];
+
+        return {
+            id: node.headNotification.notificationId,
+            uri: Uri.parse(url),
+            message: this.makeMessage(node),
+            notificationType: notificationType,
+            product: product,
+            userId: authInfo.user.id,
+            timestamp: new Date(node.headNotification.timestamp).valueOf(),
+        };
+    }
+
+    private makeMessage(node: any): string {
+        const bodyMessage = this.processBodyItems(node);
+        let message = node.headNotification.content.message + (bodyMessage ? `: ${bodyMessage}` : '');
+        // if message is too long, truncate it and add ellipsis
+        if (message.length > 200) {
+            message = message.substring(0, 200 - 3) + '...';
+        }
+        return message;
+    }
+
+    private processBodyItems(node: any): string {
+        const bodyItems = node.headNotification.content.bodyItems;
+        if (!bodyItems || bodyItems.length === 0) {
+            return '';
+        }
+
+        const bodyItem = bodyItems[0];
+        if (!bodyItem.document || bodyItem.document.format !== 'ADF') {
+            return '';
+        }
+
+        let adfData: any;
+        try {
+            adfData = JSON.parse(bodyItem.document.data);
+        } catch (error) {
+            Logger.error(new Error(`Error parsing ADF data: ${error}`));
```

Comment:
This Logger.error is not logging the original error, but it's creating a new one instead. This way we lose the original stack trace.
Also, if there are variable tokens in the error message, we will not be able to aggregate them together.

```suggestion
            Logger.error(error, 'Error parsing ADF data');
```